addons/sourcemod/scripting/surftimer/sql.sp

```diff
@@ -3458,13 +3433,359 @@ public void db_deleteCheckpoints()
 	SQL_TQuery(g_hDb, SQL_deleteCheckpointsCallback, szQuery, 1, DBPrio_Low);
 }
 
-public void SQL_deleteCheckpointsCallback(Handle owner, Handle hndl, const char[] error, any data)
-{
-	if (hndl == null)
-	{
-		LogError("[SurfTimer] SQL Error (SQL_deleteCheckpointsCallback): %s", error);
+public void SQL_deleteCheckpointsCallback(Handle owner, Handle hndl, const char[] error, any data)
+{
+	if (hndl == null)
+	{
+		LogError("[SurfTimer] SQL Error (SQL_deleteCheckpointsCallback): %s", error);
+		return;
+	}
+}
+
+public void db_viewPRinfo(int client,  char szSteamID[32], char szMapName[128])
+{
+	Handle pack = CreateDataPack();
+	WritePackCell(pack, client);
+	WritePackString(pack, szSteamID);
+	WritePackString(pack, szMapName);
+
+	if (!IsValidClient(client))
+		return;
+
+	char szQuery[1024];
+	Format(szQuery, 1024, sql_selectPR, szSteamID, szMapName, 0);
+	//PrintToConsole(client,szQuery);
+	SQL_TQuery(g_hDb, db_viewPRinfoCallback, szQuery, pack, DBPrio_Low);
+}
+
+public void db_viewPRinfoCallback(Handle owner, Handle hndl, const char[] error, any pack)
```

Comment:
`any pack` -> `DataPack pack`

Reply:
and don't forget to delete the datapack especially when you use `return` or the code will not calling `db_prinforuntimecallback` with `SQL_TQuery`, otherwise the handle will never deleted.

Reply:
the previous comment means the whole follow up, so also in the upcoming callbacks.